packages/jac/src/json-api.test.ts

```diff
@@ -127,6 +135,28 @@ describe('performs JsonApiClient request unit test', () => {
     })
   })
 
+  test('should throw custom error', async () => {
+    const api = new JsonApiClient(VALID_CFG)
+
+    const error = new CustomError('')
+    jest.spyOn(api, 'request').mockRejectedValue(error)
+
```

Comment:
this test actually does not cover your use case. maybe partly. mock response with error response (400 or some other (u could refer to the example below how to do it)) and use interceptor which will throw your custom error. 

```ts
 test('should return correct data', () => {
   const rawResponse = MockWrapper.makeFetcherResponse({
    errors: [{
       id: "1",
       code: "err_some_code",
       status: "401",
       title: "Unauthorized"
    }]
  }, 400)
   jest.spyOn(Fetcher.prototype, 'request').mockResolvedValue(rawResponse)
   const api = new JsonApiClient(VALID_CFG)
   return api.get('').then(({ data }) => expect(data).toEqual(PARSED_RESPONSE))
  })
```

Reply:
updated